include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+         */
+        enum class PublishStatus : uint8_t
+        {
+            kOk = 0,
+            kInternalError,
+            kNotAuthorized,
+            kNotAnnounced,
+            kNoSubscribers,
+            kObjectPayloadLengthExceeded,
+            kPreviousObjectTruncated,
+
+            kNoPreviousObject,
+            kObjectDataComplete,
+            kObjectContinuationDataNeeded,
+            kObjectDataIncomplete,              ///< PublishObject() was called when continuation data remains
+
+            /// Indicates that the published object data is too large based on the object header payload size plus
+            /// any/all data that was sent already.
+            kObjectDataTooLarge,
+
+            /// Previous object payload has not been completed and new object cannot start in per-group track mode
+            /// unless new group is used.
+            kPreviousObjectNotCompleteMustStartNewGroup,
+
+            /// Previous object payload has not been completed and new object cannot start in per-track track mode
+            /// without creating a new track. This requires to unpublish and to publish track again.
+            kPreviousObjectNotCompleteMustStartNewTrack,
+        };
+
+        /**
+         * @brief  Status codes for the publish track
+         */
+        enum class Status : uint8_t
+        {
+            kOK = 0,
+            kNotConnected,
+            kNotAnnounced,
+            kPendingAnnounceResponse,
+            kAnnounceNotAuthorized,
+            kNoSubscribers,
+            kSendingUnannounce,
+        };
+
+
+        // --------------------------------------------------------------------------
+        // Public API methods that normally should not be overridden
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Publish track handler constructor
+         *
+         * @param full_track_name       Full track name struct
+         * @param track_mode            The track mode to operate using
+         * @param default_priority      Default priority for objects if not specified in ObjectHeaderss
+         * @param default_ttl           Default TTL for objects if not specified in ObjectHeaderss
+         */
+        PublishTrackHandler(const FullTrackName& full_track_name,
+                            TrackMode track_mode,
+                            uint8_t default_priority,
+                            uint32_t default_ttl)
+          : BaseTrackHandler(full_track_name)
+          , track_mode_(track_mode)
+          , def_priority_(default_priority)
+          , def_ttl_(default_ttl)
+        {
+        }
+
+        // --------------------------------------------------------------------------
+        // Public Virtual API callback event methods to be overridden
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Notification of publish status
```

Comment:
this seems to be not PublishStatus field but Status field in the base class.

Reply:
It's for the publish track status, which is not the same as `PublishObjectStatus`.   

Updated the verbiage to: 

```c++
        /**
         * @brief Notification of publish track status change
         * @details Notification of a change to publish track status, such as
         *      when it's ready to publish or not ready to publish
         *
         * @param status        Indicates the status of being able to publish
         */
```